File: src/main/java/com/example/uhta/model/reciveModel/PatternModel.java
package com.example.uhta.model.reciveModel;


import com.example.uhta.model.UserModel;
import com.example.uhta.model.requestModel.AttributeModel;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Builder
@Data
@AllArgsConstructor
public class PatternModel {
    private Integer id;
    private String title;
    private String description;
    private UserModel user;
    private String date;
    private Integer frequency;
    @JsonProperty("isActive")
    private Boolean isActive;

    private List<AttributeModel> attributes;

}

File: webUi/src/models/PatternModel.ts
import { User } from "./UserModel";
import { AttributeModel } from "./AttributeModel";
import { PlateModel } from "./PlateModel";

export interface PatternModel {
  id: number;
  title: string;
  description: string;
  user: User;
  date: string;
  frequency: number;
  isActive: boolean;
  plate: PlateModel[];
  attributes: AttributeModel[];
}
